File: gerenciador-ativos-backend/src/main/java/dev/processo_seletivo/gerenciador_ativos/model/Posicao.java
package dev.processo_seletivo.gerenciador_ativos.model;

import dev.processo_seletivo.gerenciador_ativos.entity.AtivoFinanceiro;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Builder(builderMethodName = "of")
public class Posicao {

    private final AtivoFinanceiro ativoFinanceiro;
    private final BigDecimal quantidadeTotal;
    private final BigDecimal valorMercadoTotal;
    private final BigDecimal rendimento;
    private final BigDecimal lucro;

}

File: gerenciador-ativos-frontend/src/app/common/posicao.ts
import { AtivoFinanceiro } from "./ativo-financeiro";

export class Posicao {

	constructor(
		public ativoFinanceiro: AtivoFinanceiro,
		public quantidadeTotal: number,
		public valorMercadoTotal: number,
		public rendimento: number,
		public lucro: number) {}

}
